src/bootstrap/builder/tests.rs

```diff
@@ -286,7 +286,10 @@ mod dist {
         // Make sure rustdoc is only built once.
         assert_eq!(
             first(cache.all::<tool::Rustdoc>()),
-            &[tool::Rustdoc { compiler: Compiler { host: a, stage: 2 } },]
+            &[
+                tool::Rustdoc { compiler: Compiler { host: a, stage: 0 } },
+                tool::Rustdoc { compiler: Compiler { host: a, stage: 2 } }
+            ]
```

Comment:
This doesn't look right. First, the comment is now incorrect, but more importantly, it's important that we only distribute things that have been built with `--stage 2`; std uses `cfg(bootstrap)` so anything built with stage 0 can be just completely different.

I think instead you should change `UnstableBookGen` to use `builder.top_stage` instead of stage 0.

Reply:
I did this in `UnstableBookGen`, but doing this in Tidy results in stage1 being built, so I keep it stage 0 in tidy.